cdap-spark-core-base/src/main/scala/co/cask/cdap/app/runtime/spark/AbstractSparkExecutionContext.scala

```diff
@@ -215,7 +215,7 @@ abstract class AbstractSparkExecutionContext(sparkClassLoader: SparkClassLoader,
     val cConf = runtimeContext.getCConfiguration
     val tempDir = new File(cConf.get(Constants.CFG_LOCAL_DATA_DIR),
                            cConf.get(Constants.AppFabric.TEMP_DIR)).getAbsoluteFile
-    val classDir = new File(tempDir, runtimeContext.getProgramRunId + "-classes-" + interpreterCount.incrementAndGet())
+    val classDir = new File(tempDir, runtimeContext.getProgramRunId.getRun + "-classes-" + interpreterCount.incrementAndGet())
```

Comment:
This makes the directory a bit hard to relate to. Probably better use the program run id and form a string that is safe for all FS. Maybe just use `.` to separate out the parts.

Reply:
Specifically, if you turn the `toIdParts` method in `EntityId` to public, then you can simply do `runtimeContext.getProgramRunId.toIdParts.mkString(".")` in here.